src/AutoRest.CSharp/LowLevel/Output/LowLevelOutputLibraryFactory.cs

```diff
@@ -110,10 +110,19 @@ private static IReadOnlyList<ClientInfo> SetHierarchy(IReadOnlyDictionary<string
             }
 
             var topLevelClients = clientInfosByName.Values.Where(c => c.Parent == null).ToList();
-            if (!Configuration.SingleTopLevelClient || topLevelClients.Count == 1)
+            if (!Configuration.SingleTopLevelClient)
             {
                 return topLevelClients;
             }
+            if (topLevelClients.Count == 1)
+            {
+                var topClient = topLevelClients[0];
+                if (!topClient.ResourceParameters.Any() && !string.IsNullOrEmpty(topClient.OperationGroupKey))
```

Comment:
This logic is kind of duplicated. We are checking the same thing in `CreateClientInfo`.
I'd say that we should remove this check both from `SetHierarchy` and from `CreateClientInfo` and add client suffix only in the `CreateClients`, right when `LowLevelClient` isntance is instantiated.

Reply:
Hi @AlexanderSher  If we delayed to add 'client' suffix in `CreateClients`, it may have problem to apply the 'Client Hierarchy' customization because in the customization it defined the 'ParentClient' attribute, e.g. 'ParentClient = typeof(ResourceServiceClient)'. It may cause apply customization fail. We may need to think about the order this is and the way to use this attribute to construct the client hierarchy.